### ISSUE TITLE
Unknown element does not generate errors

### ISSUE THREAD

**Issue by gregoriopellegrino** [w3c/epubcheck#1655]
Hi,
while testing an EPUB, I realized that HTML code like this:

```html
<p>My paragraph with an invalid <bla-bla>bla-bla</bla-bla> element.</p>
```

The `bla-bla` element does not exist, but EPUBCheck does not report it. I tested it with Nu HTML Checker and it does not report it either. It seems that this lack of reporting is due to the presence of the hyphen in the name, which leads Nu HTML Checker to assume that the element is a custom element (without checking whether it is actually declared in JavaScript).

Gregorio

**Comment by mattgarrish** [comment]
> without checking whether it is actually declared in JavaScript

It's not required that custom elements be registered that I'm aware of. They're effectively just divs/spans if you don't, which wouldn't make them terribly useful except maybe for styling. Tag soup that reads as more intelligible, I suppose, but not invalid.

But epubcheck doesn't evaluate documents at run time, so aside from potentially flaky heuristic tests this would be impossible to verify.

**Comment by gregoriopellegrino** [comment]
In the specifications, I always find them associated with a [JavaScript statement](https://html.spec.whatwg.org/multipage/custom-elements.html), but in fact this is non-normative text.

**Comment by rdeltour** [comment]
We could always raise an informative message on such elements when no scripting was found in the document, but TBH I do not think it is worth the effort. These are valid HTML elements anyway.

I'm closing this issue, but feel free to keep on commenting or reopen if you do think EPUBCheck should be improved there.
